airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -61,14 +67,159 @@ public static Source sshWrappedSource() {
     super(DRIVER_CLASS, new MssqlJdbcStreamingQueryConfiguration(), new MssqlSourceOperations());
   }
 
+  @Override
+  public AutoCloseableIterator<JsonNode> queryTableFullRefresh(JdbcDatabase database,
+                                                               List<String> columnNames,
+                                                               String schemaName,
+                                                               String tableName) {
+    LOGGER.info("Queueing query for table: {}", tableName);
+
+    List<String> newIdentifiersList = getWrappedColumn(database,
+        columnNames,
+        schemaName, tableName, "\"");
+    String preparedSqlQuery = String
+        .format("SELECT %s FROM %s", enquoteIdentifierList(newIdentifiersList),
+            getFullTableName(schemaName, tableName));
+
+    LOGGER.info("Prepared SQL query for TableFullRefresh is: " + preparedSqlQuery);
+    return queryTable(database, preparedSqlQuery);
+  }
+
+  @Override
+  public AutoCloseableIterator<JsonNode> queryTableIncremental(JdbcDatabase database,
+                                                               List<String> columnNames,
+                                                               String schemaName,
+                                                               String tableName,
+                                                               String cursorField,
+                                                               JDBCType cursorFieldType,
+                                                               String cursor) {
+    LOGGER.info("Queueing query for table: {}", tableName);
+    return AutoCloseableIterators.lazyIterator(() -> {
+      try {
+        final Stream<JsonNode> stream = database.query(
+            connection -> {
+              LOGGER.info("Preparing query for table: {}", tableName);
+
+              final String identifierQuoteString = connection.getMetaData()
+                  .getIdentifierQuoteString();
+              List<String> newColumnNames = getWrappedColumn(database,
+                  columnNames, schemaName, tableName, identifierQuoteString);
+
+              final String sql = String.format("SELECT %s FROM %s WHERE %s > ?",
+                  getWrappedColumn(connection, newColumnNames),
```

Comment:
Is that just using `newColumnNames` will be sufficient here and the `getWrappedColumn` call is not necessary?

It is the case in `queryTableFullRefresh` above:

```java
  List<String> newIdentifiersList = getWrappedColumn(database,
        columnNames,
        schemaName, tableName, "\"");
    String preparedSqlQuery = String
        .format("SELECT %s FROM %s", enquoteIdentifierList(newIdentifiersList),
            getFullTableName(schemaName, tableName));
```

which generates the correct columns for the select clause: `SELECT "id",test_column.ToString() as "test_column" FROM "dbo"."dbo_1_hierarchyid"` because the first `getWrappedColumn` method (with 5 parameters) already adds quotation marks to the column names.

Basically I still think the second `getWrappedColumn` (with two parameters) is unnecessary.

To answer this comment:
> Unfortunately, we can not use a generic method for List as in this case we would get it enquoted like this
> "test_column.ToString() AS test_column", but need to have it like this:
> test_column.ToString() AS "test_column".
>
> So we need to have a custom enquoting for such case.

If we compare the two enquoting methods:

```java
public String getWrappedColumn(final Connection connection, final List<String> identifiers) throws SQLException {
  final StringJoiner joiner = new StringJoiner(",");
  for (final String col : identifiers) {
    if (col.contains("()")) {
      joiner.add(col);
    } else {
      final String s = sourceOperations.enquoteIdentifier(connection, col);
      joiner.add(s);
    }

  }
  return joiner.toString();
}

protected String enquoteIdentifierList(final List<String> identifiers) {
  final StringJoiner joiner = new StringJoiner(",");
  for (final String identifier : identifiers) {
    if (identifier.contains("()")) {
      joiner.add(identifier);
    } else {
      joiner.add(getIdentifierWithQuoting(identifier));
    }
  }
  return joiner.toString();
}
```

They are not doing anything differently. Specifically the identifiers won't be get enquoted like `"test_column.ToString() AS test_column"` because 1) this column will already be partially enquoted as `test_column.ToString() AS "test_column"`, and 2) any column with the `ToString()` method will be skipped and just get added to the `joiner`.

Reply:
Ohhh, now I got your point. Yup, you're absolutely right. I just worked at these parts (full refresh and incremental) if little bit different periods of time and didn't notice :) 
Updated and refactored. Thanks!